src/internal/modules/account/controller.js

```diff
@@ -56,27 +52,27 @@ const getSetPermissions = async (request, h, formFromPost) => {
 };
 
 const postSetPermissions = async (request, h) => {
-  const user = await getUser(request.params.userId);
   const { payload } = request;
+  const { userId: callingUserId } = request.defra;
+  const { newUserEmail, permission } = payload;
   const form = handleRequest(
     setPermissionsForm(request, payload),
     request,
     setPermissionsSchema
   );
 
   if (form.isValid) {
-    // TODO: Update the users permissions
-
-    // then redirect to the success page
+    const newUser = await helpers.getInternalUser(callingUserId, newUserEmail, permission);
 
```

Comment:
I think if the user has been successfully created here, this is where the session key should be deleted.
Should we be using:
```
delete request.yar.clear('key') 
```
as per the docs?